airflow/models/xcom_arg.py

```diff
@@ -0,0 +1,96 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any, Dict, List, Optional, Union
+
+from airflow.models.baseoperator import BaseOperator
+from airflow.models.xcom import XCOM_RETURN_KEY
+
+
+class XComArg:
+    """
+    Class that represents a XCom push from a previous operator.
+    Defaults to "return_value" as only key.
+    """
+
+    def __init__(self, operator: BaseOperator, keys: Optional[List[str]] = None):
+        self._operator = operator
+        self._keys = keys or [XCOM_RETURN_KEY]
+
+    def __eq__(self, other):
```

Comment:
Should this support `==` between `XcomArg` and `BaseOperator` as well?

Reply:
I would argue not. BaseOperator is a functional representation vs XComArg is a data reference, so comparing them as equal should never result in True

Reply:
Agree with @casassg